activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -358,33 +358,28 @@ def insert_fixture(fixture, table_name)
       def insert_fixtures(fixtures, table_name)
         return if fixtures.empty?
 
-        columns = schema_cache.columns_hash(table_name)
+        execute(build_fixture_sql(fixtures, table_name), "Fixtures Insert")
+      end
 
-        values = fixtures.map do |fixture|
-          fixture = fixture.stringify_keys
+      def insert_fixtures_set(fixture_set, tables_to_delete = [])
+        total_sql = "".dup
 
-          unknown_columns = fixture.keys - columns.keys
-          if unknown_columns.any?
-            raise Fixture::FixtureError, %(table "#{table_name}" has no columns named #{unknown_columns.map(&:inspect).join(', ')}.)
-          end
+        fixture_set.each do |table_name, fixtures|
+          next if fixtures.empty?
 
-          columns.map do |name, column|
-            if fixture.key?(name)
-              type = lookup_cast_type_from_column(column)
-              bind = Relation::QueryAttribute.new(name, fixture[name], type)
-              with_yaml_fallback(bind.value_for_database)
-            else
-              Arel.sql("DEFAULT")
-            end
-          end
+          total_sql << build_fixture_sql(fixtures, table_name)
+          total_sql << ";\n"
         end
 
-        table = Arel::Table.new(table_name)
-        manager = Arel::InsertManager.new
-        manager.into(table)
-        columns.each_key { |column| manager.columns << table[column] }
-        manager.values = manager.create_values_list(values)
-        execute manager.to_sql, "Fixtures Insert"
+        disable_referential_integrity do
+          transaction do
+            tables_to_delete.each { |table| delete "DELETE FROM #{quote_table_name(table)}", "Fixture Delete" }
+            packet_size = total_sql.bytesize
+            raise ActiveRecordError, "Fixtures set is too large (#{packet_size} bytes). Consider increasing the max_allowed_packet variable." if packet_too_large?(packet_size)
```

Comment:
I was hoping we could do this check a few lines above, and turn `total_sql` into an array of `< max_allowed_packet`-sized chunks.